src/NServiceBus.Core/Recoverability/SecondLevelRetries/DelayedRetryExecutor.cs

```diff
@@ -29,11 +29,11 @@ public Task Retry(IncomingMessage message, TimeSpan delay, ContextBag context)
             outgoingMessage.Headers[Headers.RetriesTimestamp] = DateTimeExtensions.ToWireFormattedString(DateTime.UtcNow);
 
             UnicastAddressTag messageDestination;
-            DeliveryConstraint[] deliveryConstraints = null;
+            List<DeliveryConstraint> deliveryConstraints = null;
             if (timeoutManagerAddress == null)
             {
                 // transport supports native deferred messages, directly send to input queue with delay constraint:
-                deliveryConstraints = new DeliveryConstraint[]
+                deliveryConstraints = new List<DeliveryConstraint>
```

Comment:
Feel free to do it while merging ;)

On Friday, July 1, 2016, Tim Bussmann <notifications@github.com> wrote:

> In
> src/NServiceBus.Core/Recoverability/SecondLevelRetries/DelayedRetryExecutor.cs
> <https://github.com/Particular/NServiceBus/pull/3893#discussion_r69269329>
> :
>
> >              if (timeoutManagerAddress == null)
> >              {
> >                  // transport supports native deferred messages, directly send to input queue with delay constraint:
> > -                deliveryConstraints = new DeliveryConstraint[]
> > +                deliveryConstraints = new List<DeliveryConstraint>
>
> since the collection initializer doesn't set the capacity property, should
> we do that here? (e.g. new List<DeliveryConstraint>(1){...})
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3893/files/79d7ac4fd8e885e7aebaec716e6efd3e289d711c#r69269329>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAKosmK-Z8I0-UBWfADCa-JN-ZLa1hifks5qRNZKgaJpZM4JCD3c>
> .
>